src/modules/evaluateExpression/evaluateExpression.ts

```diff
@@ -136,7 +141,67 @@ async function processPgSQL(queryArray: any[], queryType: string, connection: IC
   }
 }
 
-function processGraphQL(queryArray: any[], connection: object) {
-  // TO-DO -- For now just return a value that makes test work
-  return 0
+async function processGraphQL(queryArray: any[], connection: IGraphQLConnection) {
+  const query = queryArray[0]
+  const variableNames = queryArray[1]
+  const variableNodes = queryArray.slice(2, queryArray.length - 1)
+  const returnNode = queryArray[queryArray.length - 1]
+
+  const variables = zipArraysToObject(variableNames, variableNodes)
+
+  const data = await graphQLquery(query, variables, connection)
+
+  const selectedNode = extractNode(data, returnNode)
+
+  if (Array.isArray(selectedNode)) return selectedNode.map((item) => simplifyObject(item))
+  else return simplifyObject(selectedNode)
+}
+
+// Build an object from an array of field names and an array of values
+function zipArraysToObject(variableNames: string[], variableValues: any[]) {
```

Comment:
> Let's keep using only arrow functions for consistency

They're not exactly the same, though and have some (admittedly subtle) different behaviours. One of the main difference is hoisting -- arrow functions (and function expressions, generally) are not hoisted, and since I often refer to functions before they're defined, this requires some significant refactoring. Personally, I like to leave all the "helper" functions at the end of the file or block, so you can read the "main" function first before breaking down into the details, but this doesn't often work with arrow functions. For example, in the `extractNode` function, if I turn the inner function `extractNodeWithArray` into an arrow function, it no longer works unless I move the inner function to the start of the main function, which I find unintuitive to read.

(The other differences if you're interested: https://medium.com/better-programming/difference-between-regular-functions-and-arrow-functions-f65639aba256)

So yeah, for React components, I think it would make sense to stick to the arrow syntax, but I think there are cases for traditional functional declarations at times in other areas. Maybe we can have a quick chat about best practice here?

Reply:
> I understand it would be different from the way `pgSQL` is receiving the values (i.e as array), but looking at this now makes even more sense to me that, passing an object with field name and value, would be better.

Do you mean for the GraphQL operator? Yeah, that's what I tried to do initially, but then it was way harder to handle dynamic values for the field values, since it no longer fit the standard "shape" of the nodes and you'd have to treat this as a special case, which got ugly really quickly.  Let's discuss.